core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -28,12 +26,14 @@ import org.apache.kafka.common.requests.AbstractRequest
 import org.apache.kafka.common.utils.Time
 import org.apache.kafka.server.util.ShutdownableThread
 
+import java.util.Map.Entry
+import java.util.{ArrayDeque, ArrayList, Collection, Collections, HashMap, Iterator}
 import scala.jdk.CollectionConverters._
 
 /**
  *  Class for inter-broker send thread that utilize a non-blocking network client.
  */
-abstract class InterBrokerSendThread(
+class InterBrokerSendThread(
```

Comment:
Maybe we should name it InterBrokerSender?  The fact that it has only one thread is more like an internal implementation detail.